### DISCUSSION TITLE
artifact options

### DISCUSSION THREAD

**Discussion by fabrixx** [aunetx/blur-my-shell#182] [General]
What exactly are the three artifact options for?

Thanks

**Comment by aunetx** [comment]
Hello, sorry for the lonnng delay, I completely forgot about the existence of github's discussions...

The three options are to choose how should the extension manage the blur effect repaints: gnome currently has some (a lot of) bugs around the blur effect, but forcing a repaint at the right time usually manages to remove most of the effects.

The problem is then to know when to force the update: the more you repaint, the more performances you use. So I added three options:
- one (best performances) do not repaint at all, so no artifact is removed. This is good either to test, to maximize performances in some cases, ... but usually it makes the blur effect dirty.
- two (middleground) repaints when it may be needed, and now uses a little trick to catch nearly every artifact, without using too much performances
- three (best quality) repaints every time it can without degrading too much the performances, ensuring that no artifact is left over.

To be honest, I let the three options, but with the current state of the extension, artifacts options two and three are nearly the same, because the trick works really well, so using the three does nearly nothing more (but then should not use too much cpu either).

And when using static blur (which is used in overview, in panel when selected, and soon in dash-to-dock when #170 is merged, if selected), the option does nothing, as there is not a single bug with the actor blur effect (!!!), as opposed to the backdrop one used in dynamic blur :)

So, if you want to just use something without artifacts, without problem, ..., you can simply use the middle option -- and everything should work well!
